### PR TITLE
First steps in cleaning up the UI code

### PR DESCRIPTION
**Pull request by comonoidial**

### CODE REVIEW

File: Code/src/main/java/nl/utwente/viskell/ui/components/Block.java [hunk 1]
```diff
@@ -116,51 +118,65 @@ public final CustomUIPane getPane() {
...
+     * @return Only the active (as specified with by the knot index) inputs.
+     */
+    public List<InputAnchor> getActiveInputs() {
+        return getAllInputs();
+    }
+    
+    /**
+     * @return the optional output Anchor for this Block
+     */
+    public Optional<OutputAnchor> getOutputAnchor() {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Block.java]: Could consider making this

    public List<OutputAnchor> getOutputAnchors()

if we want to go N inputs/N outputs in the future.

File: Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java [hunk 2]
```diff
@@ -25,6 +32,114 @@
  */
 public abstract class ConnectionAnchor extends StackPane implements ComponentLoader {
 
+    /**
+     * Handler class that reacts to user inputs on ConnectionAnchors to be able to
+     * create, edit and drag Connections.
+     */
+    private class AnchorHandler implements EventHandler<InputEvent> {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java]: Does this have to be a nested class? Can't you just pull its methods into the ConnectionAnchor class?

File: Code/src/main/java/nl/utwente/viskell/ui/components/Connection.java [hunk 3]
```diff
@@ -226,7 +226,7 @@ private void checkErrorListener(ObservableValue<? extends Boolean> value, Boolea
     /**
      * @return Whether or not both sides of this Connection are connected to an Anchor.
      */
-    public final boolean isConnected() {
+    public final boolean isFullyConnected() {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Connection.java]: Why the name change/revert? Is there an `isHalflyConnected`?

File: Code/src/main/java/nl/utwente/viskell/ui/components/Block.java [hunk 4]
```diff
@@ -116,51 +118,65 @@ public final CustomUIPane getPane() {
...
-     * Updates the expression, clearing the expression dirty flag.
+     * Updates the expression
      */
-    public void updateExpr() {
-        setExprIsDirty(false);
-    }
+    public abstract void updateExpr();
     
-    @Override
     public final int getVisualState() {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Block.java]: These methods are now undocumented.

File: Code/src/main/java/nl/utwente/viskell/ui/components/Block.java [hunk 5]
```diff
@@ -116,51 +118,65 @@ public final CustomUIPane getPane() {
...
      * @return The expression this Block represents.
      * 
      * If the expression is not up-to-date it gets updated.
      */
-    public Expression getExpr() {
+    public final Expression getExpr() {
         // Assure expr is up-to-date.
         if (getExprIsDirty()) {
             updateExpr();
+            setExprIsDirty(false);
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Block.java]: Is this the only place that's allowed to touch the dirty flag? If so, maybe we can reduce the visibility on that flag.

File: Code/src/main/java/nl/utwente/viskell/ui/components/Connection.java (hunk 3 above)
Comment (comonoidial) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Connection.java]: isConnected exists already as unrelated internal method in javafx.
And fullyConnected is clearer (I was also trying to track down a bug with partially connected wires staying on the uipane)

File: Code/src/main/java/nl/utwente/viskell/ui/components/Block.java (hunk 5 above)
Comment (comonoidial) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Block.java]: Seems to be one other place, but cleaning up the whole getter/setter/property/visibility mess is a later step.

File: Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java (hunk 2 above)
Comment (comonoidial) [inline Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java]: Eventually it should not be an inner class anymore.
But I rather get rid of the ConnectionCreationManager first.

File: Code/src/main/java/nl/utwente/viskell/ui/components/Block.java (hunk 1 above)
Comment (comonoidial) [inline Code/src/main/java/nl/utwente/viskell/ui/components/Block.java]: Will add an todo, instead of making less obvious changes in cleanup.

**Comment by wandernauta** [comment]
Understood, thanks!

**Pull request merged**